pages/apis/graphql/graphql_resource_limits.md

```diff
@@ -0,0 +1,174 @@
+
+### Single query limit
+
+Buildkite's API has a requested complexity limit of 50,000 for each individual query. This limit is enforced prior to query execution. The intention of this limit is to prevent users from requesting an excessive number of resources in a single query.
+
+As a best practice, we recommend breaking up queries into smaller, more manageable chunks and utilizing pagination to navigate through the resulting list, rather than relying on a single large query.
+
+If the query exceeds the limit, the response will return HTTP 200 status code with the following error.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Query has complexity of 251503, which exceeds max complexity of 50000"
+    }
+  ]
+}
+```
+
+
+### Time based rate limit
+
+To ensure optimal performance, an organization can use up to 20,000 actual complexity points within a 5 minute period. By allowing a set number of actual complexity points, we provide organizations flexibility to run query of different sizes within a 5 minute window.
+
+As a best practice, we recommend utilising client size strategies like caching to lower the number of API calls, queues to schedule API calls or proper use of pagination to only request necessary data to manage time based rate limits.
+
+If an organisation exceeds the 20,000 points limit, the response will return HTTP 429 status code with the following error.
+
+```json
+{
+    "errors": [
+        {
+            "message": "Your organization has exceeded the limit of 20000 complexity points. Please try again in 187 seconds."
+        }
+    ]
+}
+```
+
+
+## Accessing limits details
+
+### HTTP headers
+
+We provide rate limit status on response headers of each GraphQL call. We have 3 rate limit related headers
+
+```js
+RateLimit-Remaining: 20
+RateLimit-Limit: 20000
+RateLimit-Reset: 120
+```
+
+`RateLimit-Remaining` provides the remaining complexity left within the current time window.  
+`RateLimit-Limit` is the complexity limit for the time window.  
+`RateLimit-Limit` is the number of seconds remaining until a new time window is started and the limits are reset.  
+
+
+### Response body
+
+You can include the header  `Buildkite-Include-Query-Stats` to the GraphQL request, which will return the complexity data along with the GraphQL response.
+
+```json
+{
+  "data" : {
+    "organization": {
+      "name": "Buildkite"
+    }
+  },
+  "stats" : {
+    "requestedComplexity": 1910,
+    "actualComplexity": 550
+  }
+}
+```
+
+## Best practices to avoid rate limit errors
+
+Designing your client application with best practices in mind is the best way to avoid throttling errors. For example, you can stagger API requests in a queue and do other processing tasks while waiting for the next queued job to run.  
+
+Consider the following best practices when designing the app:
+
+* Optimize the request by only requesting data your app requires. We recommend using specific queries rather than a single all-purpose queries.
```

Comment:
```suggestion
* Optimize the request by only requesting data your app requires. We recommend using specific queries rather than a single all-purpose query.
```